x-pack/plugins/actions/server/lib/get_oauth_access_token.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Logger } from '@kbn/core/server';
+import { GoogleAuth } from 'google-auth-library';
+import { ConnectorToken, ConnectorTokenClientContract } from '../types';
+
+interface GetOAuthJwtAccessTokenOpts {
+  connectorId?: string;
+  logger: Logger;
+  credentials: string;
+  connectorTokenClient?: ConnectorTokenClientContract;
+}
+
+export const getOAuthJwtAccessToken = async ({
+  connectorId,
+  logger,
+  credentials,
+  connectorTokenClient,
+}: GetOAuthJwtAccessTokenOpts) => {
+  let accessToken: string | null | undefined;
+  let connectorToken: ConnectorToken | null = null;
+  let hasErrors: boolean = false;
+  let credentialsJSON;
+  const expiresInSec = 3600;
+
+  if (connectorId && connectorTokenClient) {
+    // Check if there is a token stored for this connector
+    const { connectorToken: token, hasErrors: errors } = await connectorTokenClient.get({
+      connectorId,
+    });
+
+    connectorToken = token;
+    hasErrors = errors;
+  }
+
+  if (connectorToken === null || Date.parse(connectorToken.expiresAt) <= Date.now()) {
+    // Save the time before requesting token so we can use it to calculate expiration
+    const requestTokenStart = Date.now();
+
+    // Validate the service account credentials JSON file input
+    try {
+      credentialsJSON = JSON.parse(credentials);
+    } catch (error) {
+      return `Failed to parse credentials JSON file: Invalid JSON format: ${error.message ?? ''}`;
+    }
+
+    // request access token with service account credentials file
+    const auth = new GoogleAuth({
+      credentials: credentialsJSON,
+      scopes: 'https://www.googleapis.com/auth/cloud-platform',
+    });
+
+    try {
+      accessToken = await auth.getAccessToken();
+    } catch (error) {
+      return `Unable to retreive access token: ${error.message}`;
```

Comment:
```suggestion
      throw new Error(`Unable to retrieve access token: ${error.message}. Ensure Vertex AI API is enabled on cloud console.`);
```

Maybe we could add a helpful note like, `ensure Vertex AI API is enabled on cloud console` or whatever other troubleshooting steps the user can take

Reply:
<img width="678" alt="Screenshot 2024-05-29 at 10 03 42 AM" src="https://github.com/elastic/kibana/assets/6935300/7ec9d95b-e73b-4668-af92-e96bd117b95c">
This is what the error looks like if thrown as is, not very user friendly

Reply:
Required changes has been done